xtask/src/task_phd.rs

```diff
@@ -292,7 +292,20 @@ impl Cmd {
                 cmd
             }
             None => {
-                let bin = build_bin("propolis-server", propolis_args.release)?;
+                let mut server_build_env = HashMap::new();
+                server_build_env
+                    .insert("PHD_BUILD".to_string(), "true".to_string());
+                let bin = build_bin(
+                    "propolis-server",
+                    propolis_args.release,
+                    // Some PHD tests specifically cover cases where a component
+                    // in the system has encountered an error, so enable
+                    // failure-injection. Do not enable `omicron-build` like we
+                    // do in Buildomat because someone running `cargo xtask phd`
+                    // is not building a propolis-server destined for Omicron.
```

Comment:
Well, unless they are :) I think this was my use case earlier this week, and it may also come in handy if someone wants to try to reproduce a CI failure on a local machine.

Perhaps add an `omicron_build` switch to `PropolisArgs` to allow the user to opt into adding that feature?

Reply:
..... i suppose `cargo xtask phd` does build a replacement `propolis-server` doesn't it.

but yeah, a build to get `omicron-build` enabled makes a lot of sense. lemme do that.

Reply:
in writing https://github.com/oxidecomputer/propolis/pull/878/commits/8affc6504ee4fc6663dd72e2339724d41b44c969 i discovered that `cargo xtask phd run` and `cargo xtask phd run --omicron-build` end up with artifacts in different directories. so if you run one, then the other, then the first one again expecting another `propolis-server` rebuild, it'll actually just run immediately. surprise!

but the test suite passes on my workstation with or without the flag (wherein i have a reservoir configured), everything fails (as expected!) when i set the reservoir too small.